internal/dataplane/client/utils.go

```diff
@@ -166,7 +163,7 @@ func CompileTCPRouteToDataPlaneBackend(ctx context.Context, c client.Client, tcp
 			Ip:   ipint,
 			Port: gatewayPort,
 		},
-		Target: target,
+		Targets: []*Target{target},
```

Comment:
Similar to above where we said we would do TCPRoute later, maybe just link the relevant issue here in a TODO.